skbio/tree/_tree.py

```diff
@@ -3269,3 +3269,295 @@ def unpack_by_func(self, func):
                 nodes_to_unpack.append(node)
         for node in nodes_to_unpack:
             node.unpack()
+
+    @experimental(as_of="0.5.3")
+    def is_ultrametric(self):
+        """Returns whether the tree is ultrametric or not.
+
+        Returns
+        -------
+        bool
+            True is all root-to-tip distance are equal, or False if at least
+            one single distance differs
+
+        Notes
+        -----
+        This function collects all the tip to root branch lengths in a set.
+        If this set has a length of 1, then the tree is ultrametric.
```

Comment:
does this mechanism mind float rounding issues, i.e. 3.999999999 == 4.000000000 ?!